common/services/user.service.js

```diff
@@ -0,0 +1,124 @@
+"use strict";
+
+const loopback = require("loopback");
+const logger = require("../utils/logger.js");
+const moment = require("moment");
+
+const objectType = "user";
+
+exports.updateById = async (id, name, dob, address, description) => {
+  try {
+    logger.debug(
+      `In User.updateById, input id: ${id}, name: ${name}, dob: ${dob}, address: ${address}, description: ${description}.`
+    );
+    const noInforMsg = "No information input, no change.";
+    const noUserForIdMsg = `No user found for input id: ${id}.`;
+    const noInforChangedMsg =
+      "The input information are the same as in DB, no change.";
+    let result = {};
+
+    //check if no need update db.
+    if (
+      typeof name === "undefined" &&
+      typeof dob === "undefined" &&
+      typeof address === "undefined" &&
+      typeof description === "undefined"
+    ) {
+      result.statusCode = 400;
+      result.message = noInforMsg;
+      return result;
+    }
+
+    let inputLength = 0;
+    if (typeof name !== "undefined") inputLength += name.trim().length;
+    if (typeof dob !== "undefined") inputLength += dob.toString().trim().length;
+    if (typeof address !== "undefined") inputLength += address.trim().length;
+    if (typeof description !== "undefined")
+      inputLength += description.trim().length;
+    if (inputLength == 0) {
+      result.statusCode = 400;
+      result.message = noInforMsg;
+      return result;
+    }
+
+    let user = await loopback.findModel(objectType).findById(id);
+    if (!user) {
+      result.statusCode = 404;
+      result.message = noUserForIdMsg;
+      return result;
+    }
+    let updateValues = {};
+    if (user.name != name) {
+      updateValues.name = name;
+    }
+    if (!moment(user.dob).isSame(dob, "day")) {
+      updateValues.dob = dob;
+    }
+    if (user.address != address) {
+      updateValues.address = address;
+    }
+    if (user.description != description) {
+      updateValues.description = description;
+    }
+    if (JSON.stringify(updateValues) === "{}") {
```

Comment:
the validation in this function with so many `if` is tedious, 
could you find another way to make code more concise?